### PR TITLE
api: delete LoadBalancer.Helper APIs that had been deprecated for a long time

### PR DESCRIPTION
**Pull request by voidzcy**
Delete a bunch of old LoadBalancer.Helper APIs that had been deprecated for a long time.

- `createSubchannel(EquivalentAddressGroup addrs, Attributes attrs)`
`createSubchannel(List<EquivalentAddressGroup> addrs, Attributes attrs)`

    Deprecated in #5722 (v1.22)

- `updateSubchannelAddresses(Subchannel subchannel, EquivalentAddressGroup addrs)`
`updateSubchannelAddresses(Subchannel subchannel, List<EquivalentAddressGroup> addrs)`
    
    Deprecated in #5802 (v1.22)

These deprecated APIs has stronger implementation requirement for synchronization as they originally did not require invocations to be from the SynchronizationContext (although a logged warning was added later if detected the call is not from the SynchronizationContext). Their replacements relaxed the burden for implementations to be thread-safe and would throw if calls are not from the SynchronizationContext.

Other old deprecated APIs are:
- `runSerialized(Runnable task)`: deprecated in #4971 (v1.17)

- `getNameResolverFactory()`: deprecated in #5418 (1.20.0)

--------------------
TODO: will need to migrate internal usages.

### CODE REVIEW

**Comment by ejona86** [comment]
> TODO: will need to migrate internal usages.

We should migrate them before merging this PR.

**Review by ejona86 — APPROVED** [review]
CC @zhangkun83

File: netty/src/main/java/io/grpc/netty/ProtocolNegotiators.java [hunk 1]
```diff
@@ -397,14 +396,6 @@ public void handlerAdded(ChannelHandlerContext ctx) throws Exception {
       ctx.pipeline().addBefore(ctx.name(), /* name= */ null, this.executor != null
           ? new SslHandler(sslEngine, false, this.executor)
           : new SslHandler(sslEngine, false));
-
-      // Support exporting the key and session identifier to the log named
-      // "io.netty.wireshark" when the system property named "io.netty.ssl.masterKeyHandler"
-      // is "true". This feature is used to analyze gRPC traffic with Wireshark.
-      if (Boolean.getBoolean(SslMasterKeyHandler.SYSTEM_PROP_KEY)) {
```
Comment (dapengzhang0) [inline netty/src/main/java/io/grpc/netty/ProtocolNegotiators.java:404]: Why this is deleted?

Comment (voidzcy) [reply netty/src/main/java/io/grpc/netty/ProtocolNegotiators.java:404]: Oh, it had some problem merging with the master. Fixing it now.

**Review by dapengzhang0 — APPROVED** [review]

**Pull request merged**